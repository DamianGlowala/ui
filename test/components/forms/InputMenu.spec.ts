import { describe, it, expect } from 'vitest'
import { UInputMenu } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('InputMenu', () => {
  it.each([
    ['basic case', {}],
    ['with custom name and placeholder', { props: { name: 'username', placeholder: 'Enter your username' } }],
    ['with disabled', { props: { disabled: true } }],
    ['with required', { props: { required: true } }],
    ['with custom class', { props: { class: 'custom-class' } }],
    ['with custom style', { props: { style: { color: 'red', fontSize: '16px' } } }],
    ['with custom type', { props: { type: 'password' } }],
    ['with nullable', { props: { nullable: true } }],
    ['with query', { props: { query: 'searchQuery' } }],
    ['with by', { props: { by: 'name' } }],
    ['with options', { props: { options: ['Option 1', 'Option 2', 'Option 3'] } }],
    ['with id', { props: { id: 'comboboxId' } }],
    ['with leading', { props: { leading: true } }],
    ['with trailing', { props: { trailing: true } }],
    ['with loading', { props: { loading: true } }],
    ['with selectedIcon', { props: { selectedIcon: 'check' } }],
    ['with optionAttribute', { props: { optionAttribute: 'value' } }],
    ['with valueAttribute', { props: { valueAttribute: 'id' } }],
    ['with searchAttributes', { props: { searchAttributes: ['name', 'description'] } }],
    ['with debounce', { props: { debounce: 300 } }],
    ['with popper', { props: { popper: { key: { id: '1' }, placement: 'bottom', arrow: true } } }],
    ['with inputClass', { props: { inputClass: 'input-class' } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof InputMenu.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, UInputMenu)
    expect(html).toMatchSnapshot()
  })
})
