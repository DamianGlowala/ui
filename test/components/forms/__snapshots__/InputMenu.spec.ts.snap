// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputMenu > renders basic case correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_0" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_1">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_1" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with by correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_20" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_21">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_21" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with custom class correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_10" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_11">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_11" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with custom name and placeholder correctly 1`] = `
"<input style="position: fixed; height: 0px; padding: 0px; overflow: hidden; clip: rect(0, 0, 0, 0); white-space: nowrap; border-width: 0px; display: none;" type="hidden" hidden="" readonly="" name="username">
<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_2" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" name="username" placeholder="Enter your username" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_3">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_3" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with custom style correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_12" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" style="color: red; font-size: 16px;" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_13">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_13" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with custom type correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_14" aria-expanded="false" aria-autocomplete="list" role="combobox" type="password" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_15">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_15" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with debounce correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_39" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_40">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_40" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with disabled correctly 1`] = `
"<div class="" data-headlessui-state="disabled">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_6" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" disabled="" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_7">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_7" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" disabled="" data-headlessui-state="disabled" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with id correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="comboboxId" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_24">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_24" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with inputClass correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_43" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="input-class" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_44">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_44" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with leading correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_25" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_26">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_26" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with loading correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_29" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_30"><span class="absolute inset-y-0 start-0 flex items-center pointer-events-none px-2.5"><span class="i-heroicons-arrow-path-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5 animate-spin"></span></span><button id="headlessui-combobox-button-PoJY0fGVvn_30" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button></div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with nullable correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_16" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_17">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_17" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with optionAttribute correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_33" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_34">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_34" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with options correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_22" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_23">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_23" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with popper correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_41" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_42">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_42" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with query correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_18" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_19">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_19" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with required correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_8" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" required="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_9">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_9" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with searchAttributes correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_37" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_38">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_38" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with selectedIcon correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_31" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_32">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_32" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with trailing correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_27" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_28">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_28" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with value correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_4" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_5">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_5" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;

exports[`InputMenu > renders with valueAttribute correctly 1`] = `
"<div class="" data-headlessui-state="">
  <div class=""><input value="" id="headlessui-combobox-input-PoJY0fGVvn_35" aria-expanded="false" aria-autocomplete="list" role="combobox" type="text" tabindex="0" data-headlessui-state="" class="" autocomplete="off" aria-labelledby="headlessui-combobox-button-PoJY0fGVvn_36">
    <!--v-if--><button id="headlessui-combobox-button-PoJY0fGVvn_36" type="button" tabindex="-1" aria-haspopup="listbox" aria-expanded="false" data-headlessui-state="" class="absolute inset-y-0 end-0 flex items-center px-2.5"><span class="i-heroicons-chevron-down-20-solid flex-shrink-0 text-gray-400 dark:text-gray-500 h-5 w-5"></span></button>
  </div>
  <!--v-if-->
</div>"
`;
